staging/src/k8s.io/kubectl/pkg/cmd/cp/cp.go

```diff
@@ -247,15 +246,16 @@ func (o *CopyOptions) copyToPod(src, dest fileSpec, options *exec.ExecOptions) e
 	}
 	reader, writer := io.Pipe()
 
-	// strip trailing slash (if any)
-	if dest.File != "/" && strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/") {
+	// strip trailing slash (if any) both unix and windows style
+	if (dest.File != "/" && strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/")) ||
+		(dest.File != "\\" && strings.HasSuffix(string(dest.File[len(dest.File)-1]), "\\")) {
 		dest.File = dest.File[:len(dest.File)-1]
 	}
 
 	if err := o.checkDestinationIsDir(dest); err == nil {
 		// If no error, dest.File was found to be a directory.
 		// Copy specified src into it
-		dest.File = dest.File + "/" + path.Base(src.File)
+		dest.File = filepath.Join(dest.File, filepath.Base(src.File))
```

Comment:
in `copyToPod`, isn't our destination assumed to be a linux container? if so, won't this set dest.File to windows path when the `cp` command is run on windows, and break the command constructed below?

Reply:
Theoretically - yes, but we need to ensure that if either is somehow injected we can fix that. I'll actually revert my changes to work only with linux style separator and will add replacing at the beginning of this function.